onnxruntime/core/providers/cuda/cuda_allocator.cc

```diff
@@ -77,6 +77,20 @@ void* CUDAExternalAllocator::Alloc(size_t size) {
 
 void CUDAExternalAllocator::Free(void* p) {
   free_(p);
+  std::lock_guard<OrtMutex> lock(lock_);
+  auto it = reserved_.find(p);
+  if (it != reserved_.end()) {
+    reserved_.erase(it);
+    if (empty_cache_) empty_cache_();
+  }
```

Comment:
if we don't find it in `reserved_`, it is a failure and points to some internal error ? Should there be handling of that instead of silently returning ?

Reply:
CUDAExternalAllocator::Free is for both Alloc and Reserve. At the beginning of the function, free_(p) handle the memory free. If we find it in the set, it's reserve case and we remove the pointer from the set. Do we really need to handle the case that the pointer points to some internal bug? The code before the change just doesn't handle such case. Do we need to add another set of non-reserved pointer for this?

Reply:
Sure, we can skip it for now